samtranslator/schema/schema.py

```diff
@@ -12,6 +12,8 @@
 # Value passed directly to CloudFormation; not used by SAM
 PassThrough = Any  # TODO: Make it behave like typescript's unknown
 
+ResourcePolicy = Union[str, Dict[str, Any]]
```

Comment:
nit: `ResourcePolicy` sounds more abstract than it is (a dict or string) and can't really be reused

Reply:
Actually I don't like this name, since now all the ones in `ResourcePolicy` have this type, but it's not obvious that they all should share the same type (whatever it is).